cosmic-core/api/src/test/java/com/cloud/api/command/admin/user/GetUserCmdTest.java

```diff
@@ -0,0 +1,193 @@
+package com.cloud.api.command.admin.user;
+
+import static org.mockito.Mockito.when;
+
+import com.cloud.api.ResponseGenerator;
+import com.cloud.api.response.UserResponse;
+import com.cloud.context.CallContext;
+import com.cloud.user.AccountService;
+import com.cloud.user.User;
+import com.cloud.user.UserAccount;
+import com.cloud.utils.exception.InvalidParameterValueException;
+
+import java.util.Date;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class GetUserCmdTest {
+    public static final Logger s_logger = LoggerFactory.getLogger(GetUserCmdTest.class.getName());
+
+    @Mock
+    private AccountService accountService;
+
+    @Mock
+    private ResponseGenerator responseGenerator;
+
+    @InjectMocks
+    private final GetUserCmd getUserCmd = new GetUserCmd();
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        CallContext.unregister();
+    }
+
+    @Test
+    public void testExecuteWithUnknownApikey() {
+        UserAccount userAccount = null;
+
+        when(accountService.getUserByApiKey(null)).thenReturn(userAccount);
+
+        try {
+            getUserCmd.execute();
+            Assert.fail("should throw exception for user/apikey not found");
+        } catch( final InvalidParameterValueException e) {
+            Assert.assertTrue("Received exception as the mock getUserByApiKey returns null result", true);
+        }
+    }
+
+    @Test
+    public void testExecuteWithDefault() {
+        final String secretKey = "Very secret key";
+        UserResponse response = new UserResponse();
+        UserAccount userAccount = new UserAccount() {
```

Comment:
Mock it with mockito :) http://www.vogella.com/tutorials/Mockito/article.html#creating-mock-objects-with-mockito